ticker_service.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"errors"
+	"log"
+	"sync"
+	"time"
+)
+
+type TickerService struct {
+	mu           sync.Mutex
+	ds           *DeviceService
+	measurements chan<- Measurement
+	stop         chan bool
+	isRunning    bool
+}
+
+func (ts *TickerService) Start() error {
+	ts.mu.Lock()
+	defer ts.mu.Unlock()
+
+	if ts.isRunning {
+		log.Println("tickers started already")
+		return nil
+	}
+
+	devices, err := ts.ds.GetAll(0, 0)
+	if err != nil {
+		return errors.New("failed to start measurements sending")
+	}
+
+	ts.stop = make(chan bool)
+	ts.isRunning = true
+
+	for _, device := range devices {
+		ts.createTickerForDevice(device)
+	}
+
+	return nil
+}
+
+func (ts *TickerService) Stop() {
+	ts.mu.Lock()
+	defer ts.mu.Unlock()
+
+	if ts.isRunning {
+		log.Println("stopping send measurements")
+		close(ts.stop)
+		ts.isRunning = false
+	}
+}
+
+func (ts *TickerService) NotifyDeviceCreated(device Device) {
+	ts.mu.Lock()
+	defer ts.mu.Unlock()
+
+	if !ts.isRunning {
+		log.Println("measurements sending not run")
+		return
+	}
+
+	ts.createTickerForDevice(device)
+}
+
+func (ts *TickerService) createTickerForDevice(device Device) {
+	ticker := time.NewTicker(time.Second * time.Duration(device.Interval))
+
+	go func(ticker *time.Ticker, deviceId int, value float64) {
```

Comment:
good practive is to let the use of you method decide if the code should be run synchronously or in goroutine. This can help with testing or reusing the method in different contexts. Of course there are exceptions - sometimes the metod suggests that it will do something in goroutine (like `runAsync`).

In this case I would just move it out of anonymous func and call it like `go ts.createTickerForDevice(device)`